pyopia/process.py

```diff
@@ -195,7 +195,7 @@ def write_segmented_images(imbw, imc, settings, timestamp):
 
 
 def extract_particles(imc, timestamp, Classification, region_properties,
-                      export_outputpath=None, min_length=0):
+                      export_outputpath=None, min_length=0, additional_propnames=None):
```

Comment:
I think it would be better to define propnames as an optional input with the default value here like this:

`def extract_particles(imc, timestamp, Classification, region_properties,
                      export_outputpath=None, min_length=0):
                      export_outputpath=None, min_length=0,
                      propnames=['major_axis_length', 'minor_axis_length', 'equivalent_diameter', 'solidity']`):

This way, users will see in the function def what is already defined, and the code on line 246 and below would not be needed.

Reply:
one silly thing with my suggestion is that users could potentially define propnames that don't have feks. 'equivalent_diameter', which would be critical for most future calculations. so to secure the minimum list keys might need some definition of these which could be extended in a similar way to you PR.....

maybe @nepstad has some ideas relating to this?

Reply:
alternately, those default propnames could be included in the function docstring, so it is then clear what is there already. Otherwise we'd need that check and extension. Is more than equivalent_diameter needed?

Reply:
yes - we could write in the docstring that propnames=['major_axis_length', 'minor_axis_length', 'equivalent_diameter'] is always required, but that additions can be added based on the skimage regionprops documentation. People would then get an error if they didn't define these properly. we could add some assertions as a check (which would still be easier code to follow I think, as we wouldn't need to pass the additional_propnames around)